Dockerfile

```
@@ -75,10 +117,21 @@ RUN ldconfig -v
 
 COPY ./t /tmp/t
 
+# nginx-tests has no upstream releases, so it is pinned to an immutable commit
+# SHA (GitHub source tarballs are byte-stable per commit) + verified sha256,
+# rather than the mutable `tip` the previous revision fetched. Keep in sync with
+# NGINX_TESTS_REF / NGINX_TESTS_SHA256 in .github/versions.env.
+ARG NGINX_TESTS_REF=76bb761cdcaeb6cb024d605c2ef91bf8d8a602d7
+ARG NGINX_TESTS_SHA256=d67d4237799cd3c9509f2555336fed65f3ceb3a6266f8caf71842cac14bb40cd
+
 RUN apt-get update -qq && \
```

Hadolint (2.15.1)

[warning] Use WORKDIR to switch to a directory

(DL3003)